framework/src/modules/chain/blocks/block_reward.js

```diff
@@ -38,117 +38,101 @@ const parseHeight = height => {
+	height = parseHeight(height);
+	const distance = Math.floor(blockRewardArgs.distance);
+	let supply = new BigNum(blockRewardArgs.totalAmount);
 
-	/**
-	 * Description of the function.
-	 *
-	 * @param {number} height
-	 * @returns {Bignumber}
-	 * @todo Add description for the function, params and the return value
-	 */
-	calcReward(height) {
-		height = parseHeight(height);
-
-		if (height < this.rewardOffset) {
-			return new BigNum(0);
-		}
-		return new BigNum(this.milestones[this.calcMilestone(height)]);
+	if (height < blockRewardArgs.rewardOffset) {
+		// Rewards not started yet
+		return supply;
 	}
 
-	/**
-	 * Description of the function.
-	 *
-	 * @param {number} height
-	 * @returns {Bignumber}
-	 * @todo Add description for the function, params and the return value
-	 */
-	calcSupply(height) {
-		height = parseHeight(height);
-		let supply = new BigNum(this.totalAmount);
-
-		if (height < this.rewardOffset) {
-			// Rewards not started yet
-			return supply;
-		}
+	const milestone = calcMilestone(height, blockRewardArgs);
+	const rewards = [];
 
-		const milestone = this.calcMilestone(height);
-		const rewards = [];
+	let amount = 0;
+	let multiplier = 0;
 
-		let amount = 0;
-		let multiplier = 0;
+	// Remove offset from height
+	height -= blockRewardArgs.rewardOffset - 1;
 
-		// Remove offset from height
-		height -= this.rewardOffset - 1;
+	for (let i = 0; i < blockRewardArgs.milestones.length; i++) {
+		if (milestone >= i) {
+			multiplier = blockRewardArgs.milestones[i];
 
-		for (let i = 0; i < this.milestones.length; i++) {
-			if (milestone >= i) {
-				multiplier = this.milestones[i];
-
-				if (height < this.distance) {
-					// Measure this.distance thus far
-					amount = height % this.distance;
-				} else {
-					amount = this.distance; // Assign completed milestone
-					height -= this.distance; // Deduct from total height
+			if (height < distance) {
+				// Measure distance thus far
+				amount = height % distance;
+			} else {
+				amount = distance; // Assign completed milestone
+				height -= distance; // Deduct from total height
 
-					// After last milestone
-					if (height > 0 && i === this.milestones.length - 1) {
-						amount += height;
-					}
+				// After last milestone
+				if (height > 0 && i === blockRewardArgs.milestones.length - 1) {
+					amount += height;
 				}
-
-				rewards.push([amount, multiplier]);
-			} else {
-				break; // Milestone out of bounds
 			}
-		}
 
-		for (let i = 0; i < rewards.length; i++) {
-			const reward = rewards[i];
-			supply = supply.plus(new BigNum(reward[0]).times(reward[1]));
+			rewards.push([amount, multiplier]);
+		} else {
+			break; // Milestone out of bounds
 		}
+	}
 
-		return supply;
+	for (let i = 0; i < rewards.length; i++) {
+		const reward = rewards[i];
+		supply = supply.plus(new BigNum(reward[0]).times(reward[1]));
 	}
-}
 
-module.exports = { BlockReward };
+	return supply;
+};
+
+module.exports = { calcMilestone, calcSupply, calcReward };
```

Comment:
```suggestion
module.exports = { calculateMilestone, calculateSupply, calculateReward };
```

Reply:
Yeah the names were not nice, I changed the names as per your suggestion